segmentation.py

```diff
@@ -0,0 +1,575 @@
+
+    def __init__(self, mu, omega, sigma, eta):
+
+        # Get size of data.
+        m, n = np.shape(mu)
+        self.__m, self.__n = m, n
+
+        # Check that the location parameter is a matrix of finite numbers.
+        if not (np.ndim(mu) == 2 and
+                np.shape(mu) == (m, n) and
+                not np.isnan(mu).any() and
+                np.isfinite(mu).all()):
+            msg = 'The location parameter must be a matrix of finite numbers.'
+            raise Exception(msg)
+
+        # Check that the scale parameter is a symmetric, positive-definite
+        # matrix.
+        if not (np.ndim(omega) == 2 and
+                np.shape(omega) == (m, m) and
+                not np.isnan(omega).any() and
+                np.isfinite(omega).all() and
+                np.allclose(np.transpose(omega), omega) and
+                linalg.det(omega) > 0.0):
+            msg = 'The scale parameter must be a symmetric, positive-definite'
+            msg += ' matrix.'
+            raise Exception(msg)
+
+        # Check that the dispersion parameter is a symmetric, positive-definite
+        # matrix.
+        if not (np.ndim(sigma) == 2 and
+                np.shape(sigma) == (n, n) and
+                not np.isnan(sigma).any() and
+                np.isfinite(sigma).all() and
+                np.allclose(np.transpose(sigma), sigma) and
+                linalg.det(sigma) > 0.0):
+            msg = 'The noise parameter must be a symmetric, positive-definite'
+            msg += ' matrix.'
+            raise Exception(msg)
+
+        # Check that the shape parameter is a number greater than one minus the
+        # number of degrees of freedom.
+        if not (np.isscalar(eta) and
+                not np.isnan(eta) and
+                np.isfinite(eta) and eta > n - 1.0):
+            msg = 'The shape parameter must be greater than one minus the'
+            msg += ' degrees of freedom.'
+            raise Exception(msg)
+
+        # Allocate space for storing the matrix of product statistics.
+        self.__prod = np.zeros([m + n, m + n])
+
+        # Initialize the statistics with the parameters of the prior
+        # distribution.
+        x = np.dot(omega, mu)
+        self.__prod[:m, :m] = omega
+        self.__prod[:m, m:] = x
+        self.__prod[m:, :m] = x.transpose()
+        self.__prod[m:, m:] = np.dot(np.transpose(mu), x) + eta * sigma
+        self.__weight = eta
+
+    def update(self, X, Y):
+        """Update the sufficient statistics given observed data.
+
+        The sufficient statistics represent the only parameters required to
+        describe the shape of the distribution. Due to conjugacy, the
+        sufficient statistics also form the hyper-parameters of the
+        distribution. In a Bayesian context, when the sufficient statistics are
+        referred to as 'hyper-parameters', it implies the sufficient statistics
+        reflect prior information about the distribution. After updating the
```

Comment:
I think the mixup is mainly due to wording. When you say hyper-parameters, you're usually talking about some Bayesian model. This Bayesian model has some prior distribution over its parameters. The *parameters* of this prior are the *hyper-parameters* of the Bayesian model.

In this case, the `MatrixGaussianInverseGamma` class doesn't know about the `ChangeDetector` class. Therefore, in this context I find it more appropriate (and less confusing) to talk about `MatrixGaussianInverseGamma`'s *parameters* (which are `ChangeDetector`'s *hyper-parameters*).

Does this make sense?